Add few Reference use cases (#31)
ietf-rats/draft-deshpande-multi-verifier: draft-deshpande-rats-multi-verifier.md

5. The mix today is a combination of Verifier services provided by component manufacturers, Verifiers provided by integrators, and Verifiers under local authority (i.e., close to the attester). Rarely is it just one of these.

# Reference Use Cases

> This section covers generic use cases that demonstrate Multi Verifier applicability, independent of specific solutions.
> The purpose is to provide motivation for various aspects of the architecture presented in this document.
> Many other use cases exist; this document does not contain a complete list.

Reviewer: ```suggestion
This section covers generic use cases that demonstrate the applicability of Multi Verifier, regardless of specific solutions.
Its purpose is to motivate various aspects of the architecture presented in this document.
There are many other use cases; this document does not contain a complete list.
```

## Verification of Devices containing heterogenous components
A typical modern Device contains not only a traditional Central Processing Unit (CPU), but also heterogeneous acceleration components (GPU, NPU, TPU, etc) from different suppliers, to acheive a specific purpose.

> For example, either to speed up the processing or to assist in Inferencing. The trustworthiness assessment of the Device requires trust in all these individual components.

Author: ```suggestion
For example, either to speed up the processing or to assist in AI Inference. The trustworthiness assessment of the Device requires trust in all these individual components.
```

> However, the Verifier for each type of component may be deployed by each vendor separately, and these vendors may prefer to keep the Attestation Verification inputs or the Verification algorithm seperate and isolated from other Verifiers in the eco-system, due to business concerns.

Author: ```suggestion
However, the Verifier for each type of component may be deployed by each vendor separately, and these vendors may prefer to keep the Attestation Verification inputs or the Verification algorithm separate and isolated from other Verifiers in the eco-system, due to business concerns, such as complexity or cost.
```

> When such a Verifier operates, it needs to interact with other Verifiers and hence needs to understand the topology and information exchange, for example to receive partial Evidence, specific to the relevant component or convey partial Attestation Results for the component.

Author: ```suggestion
When these Verifiers operate in concert, they need to interact with each other and hence needs to understand the topology and need to inter work using standardised protocols, for example to receive partial Evidence, specific to the relevant component or convey partial Attestation Results for the component.
```

> A typical Device contains not only a traditional Central Processing Unit (CPU), but also heterogeneous acceleration components (GPU, NPU, TPU, etc) from different suppliers, to acheive a specific purpose.
> 
> For example, either to speed up the processing or to assist in AI Inference. The trustworthiness assessment of the Device requires trust in all these individual components.
> However, the Verifier for each type of component may be deployed by each vendor separately, and these vendors may prefer to keep the Attestation Verification inputs or the Verification algorithm separate and isolated from other Verifiers in the eco-system, due to business concerns, such as complexity or cost.
> 
> When these Verifiers operate in concert, they need to interact with each other and hence needs to understand the topology and need to inter work using standardised protocols, for example to receive partial Evidence, specific to the relevant component or convey partial Attestation Results for the component.

Reviewer: ```suggestion
A device may contain a central processing unit (CPU), as well as heterogeneous acceleration components (such as GPUs, NPUs and TPUs) from different suppliers.

These components can be used to speed up processing or assist with AI inference.
Trustworthiness assessment of the device requires trust in all of these components.
However, due to business concerns such as scalability, complexity and cost of infrastructure, the verifier for each type of component may be deployed separately by each vendor.

When these Verifiers operate together, they must interact with each other, understand the topology and interoperate using standardised protocols. For instance, they may need to exchange partial evidence relating to the relevant component or partial attestation results for it.
```

> Attester: A modern Device having mulitple components

Author: ```suggestion
Attester: A Composite Device having multiple components
```

> When these Verifiers operate in concert, they need to interact with each other and hence needs to understand the topology and need to inter work using standardised protocols, for example to receive partial Evidence, specific to the relevant component or convey partial Attestation Results for the component.

Reviewer: ```suggestion
A device may contain a central processing unit (CPU), as well as heterogeneous acceleration components (such as GPUs, NPUs and TPUs) from different suppliers.

These components can be used to speed up processing or assist with AI inference.
Trustworthiness assessment of the device requires trust in all of these components.
However, due to business concerns such as scalability, complexity and cost of infrastructure, the verifier for each type of component may be deployed separately by each vendor.

When these Verifiers operate together, they must interact with each other, understand the topology and interoperate using standardised protocols. For instance, they may need to exchange partial evidence relating to the relevant component or partial attestation results for it.
```

> Relying Party: An entity which is making trust based decisions for such an Attester

Author: ```suggestion
Relying Party: An entity which is making trust decisions for such an Attester
```

## Verification of Workloads operating in Confidential Computing environment

> Confidential Computing is becoming increasingly important as organizations move more workloads into untrusted or shared environments. In such a system the application or workload (could be an AI model, database process, financial service, etc.) is prepared to run inside a TEE-enabled VM or an enclave. When the workload starts, the TEE generates a cryptographic attestation report proving:

Author: ```suggestion
Confidential Computing is becoming increasingly important as organisations move more workloads into untrusted or shared environments. In such a system the application or workload (could be an AI model, database process, financial service, etc.) is executed inside a TEE-protected VM. When the workload starts, the TEE generates a cryptographic attestation report providing:
```

1. The workload is running on a platform with a known state.

> 2. The workload is running correct application.

Author: ```suggestion
2. The workload is running correct application(s).
```

> 2. The workload is running correct application with a known set of claims pertaining to the workload.

Author: ```suggestion
1. The workload is running on a platform with a known state.
2. The workload is running correct application.

The platform is often built by an independent TEE Vendor, while the Workloads are deployed by Workload owners that come from different parts of supply chain.

The Verification of Attestation of such a system requires independent but mutually coordinating verification of
platform claims  appraised by a Platform Verifier and Workload Claims appraised by a Workload Verifier.

Attester: A layered Attester containing a platform and a workload running in a CC environment

> Relying Party: An entity which is making trust based decisions, such as a key release to a workload

Author: ```suggestion
Relying Party: An entity which is making trust decisions, such as a key release to a workload
```
